web/frontend/src/pages/form/components/utils/useQuestionForm.ts

```diff
@@ -56,38 +80,113 @@ const useQuestionForm = (initState: RankQuestion | SelectQuestion | TextQuestion
       }
     };
 
-  // updates the choices array when the user adds a new choice
-  const addChoice = () => {
-    setState({ ...state, Choices: [...Choices, ''], MaxN: Choices.length + 1 });
+  // updates the ChoicesMap array when the user adds a new choice
+  const addChoice = (lang) => {
+    const obj = Object.fromEntries(ChoicesMap);
+    const newChoicesMap = new Map(Object.entries(obj));
+    switch (lang) {
+      case 'en':
+        setState({
+          ...state,
+          ChoicesMap: newChoicesMap.set('en', [...newChoicesMap.get('en'), '']),
+          MaxN: ChoicesMap.get('en').length + 1,
+        });
+        break;
+      case 'fr':
+        setState({
+          ...state,
+          ChoicesMap: newChoicesMap.set('fr', [...newChoicesMap.get('fr'), '']),
+          MaxN: ChoicesMap.get('fr').length + 1,
+        });
+        break;
+      case 'de':
+        setState({
+          ...state,
+          ChoicesMap: newChoicesMap.set('de', [...newChoicesMap.get('de'), '']),
+          MaxN: ChoicesMap.get('de').length + 1,
+        });
+        break;
+      default:
+        setState({
+          ...state,
+          ChoicesMap: newChoicesMap.set('en', [...newChoicesMap.get('en'), '']),
+          MaxN: ChoicesMap.get('en').length + 1,
+        });
+    }
   };
 
-  // remove a choice from the choices array
+  // remove a choice from the ChoicesMap array
   const deleteChoice = (index: number) => {
-    if (Choices.length > MinN) {
-      const filteredChoices = Choices.filter((item: string, idx: number) => idx !== index);
-
+    if (ChoicesMap.get('en').length > MinN) {
+      const filteredChoicesMap = ChoicesMap.get('en').filter(
+        (item: string, idx: number) => idx !== index
+      );
+      setState({
```

Comment:
this is a lot of duplication. You can easily create a function that updates the list at a given language key in the map. Also, you should do the state update only once, as updating a state triggers a re-render and can be expensive.